src/cli.rs

```diff
@@ -0,0 +1,764 @@
+    /// limit on the search depth.
+    #[clap(
+        long,
+        short = 'd',
+        value_name = "depth",
+        value_parser,
+        alias("maxdepth")
+    )]
+    max_depth: Option<usize>,
+    /// Only show results starting at given depth
+    ///
+    /// Only show search results starting at the given depth.
+    /// See also: '--max-depth' and '--exact-depth'
+    #[clap(long, value_name = "depth", hide_short_help = true, value_parser)]
+    min_depth: Option<usize>,
+    /// Only show results at exact given depth
+    ///
+    /// Only show search results at the exact given depth. This is an alias for
+    /// '--min-depth <depth> --max-depth <depth>'.
+    #[clap(long, value_name = "depth", hide_short_help = true, value_parser, conflicts_with_all(&["max-depth", "min-depth"]))]
+    exact_depth: Option<usize>,
+    /// Do not travers into matching directories
+    ///
+    /// Do not traverse into directories that match the search criteria. If
+    /// you want to exclude specific directories, use the '--exclude=…' option.
+    #[clap(long, hide_short_help = true, action, conflicts_with_all(&["size", "exact-depth"]))]
+    pub prune: bool,
+    /// Filter by type: file (f), directory (d), symlink (l),
+    /// executable (x), empty (e), socket (s), pipe (p)
+    ///
+    /// Filter the search by type:
+    ///
+    ///   'f' or 'file':         regular files
+    ///   'd' or 'directory':    directories
+    ///   'l' or 'symlink':      symbolic links
+    ///   's' or 'socket':       socket
+    ///   'p' or 'pipe':         named pipe (FIFO)
+    ///
+    ///   'x' or 'executable':   executables
+    ///   'e' or 'empty':        empty files or directories
+    ///
+    /// This option can be specified more than once to include multiple file types.
+    /// Searching for '--type file --type symlink' will show both regular files as
+    /// well as symlinks. Note that the 'executable' and 'empty' filters work differently:
+    /// '--type executable' implies '--type file' by default. And '--type empty' searches
+    /// for empty files and directories, unless either '--type file' or '--type directory'
+    /// is specified in addition.
+    ///
+    /// Examples:
+    ///
+    ///   - Only search for files:
+    ///       fd --type file …
+    ///       fd -tf …
+    ///   - Find both files and symlinks
+    ///       fd --type file --type symlink …
+    ///       fd -tf -tl …
+    ///   - Find executable files:
+    ///       fd --type executable
+    ///       fd -tx
+    ///   - Find empty files:
+    ///       fd --type empty --type file
+    ///       fd -te -tf
+    ///   - Find empty directories:
+    ///       fd --type empty --type directory
+    ///       fd -te -td"
+    #[clap(long = "type", short = 't', value_name = "filetype", hide_possible_values = true,
+        arg_enum, action = ArgAction::Append, number_of_values = 1, verbatim_doc_comment)]
```

Comment:
Does this need `action = ArgAction::Append, number_of_values = 1`?

ditto for all of the others